homeassistant/components/syncthing/config_flow.py

```diff
@@ -0,0 +1,82 @@
+"""Config flow for syncthing integration."""
+import logging
+
+import aiosyncthing
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import CONF_TOKEN, CONF_URL, CONF_VERIFY_SSL
+from homeassistant.util.network import normalize_url
+
+from .const import DEFAULT_URL, DEFAULT_VERIFY_SSL, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_URL, default=DEFAULT_URL): str,
+        vol.Required(CONF_TOKEN): str,
+        vol.Required(CONF_VERIFY_SSL, default=DEFAULT_VERIFY_SSL): bool,
+    }
+)
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect."""
+
+    for entry in hass.config_entries.async_entries(DOMAIN):
+        if entry.data[CONF_URL] == normalize_url(data[CONF_URL]):
```

Comment:
Can we use the server id as unique id for the config entry?

https://developers.home-assistant.io/docs/config_entries_config_flow_handler#unique-id-requirements